src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx

```diff
@@ -157,57 +157,69 @@ const ContractInteractionSection = ({
   const onContractChange = useCallback(
     (contract: AnyUser) => {
       setIsLoadingABI(true);
-      setFetchABIError('');
-      if (selectedSafe && isAddress(contract.profile.walletAddress)) {
+      setABIError('');
+      if (selectedVerifiedSafe && isAddress(contract.profile.walletAddress)) {
         const contractPromise = fetchContractABI(
           contract.profile.walletAddress,
-          Number(selectedSafe.chainId),
+          Number(selectedVerifiedSafe.chainId),
         );
         contractPromise.then((data) => {
           setIsLoadingABI(false);
           if (data) {
             onContractABIChange(data);
           } else {
-            setFetchABIError(MSG.fetchFailedError);
+            setABIError(MSG.fetchFailedError);
           }
         });
       } else {
         if (!isAddress(contract.profile.walletAddress)) {
-          setFetchABIError(MSG.invalidAddressError);
+          setABIError(MSG.invalidAddressError);
         } else {
-          setFetchABIError(MSG.noSafeSelectedError);
+          setABIError(FundsMSG.invalidSafeError);
         }
         setIsLoadingABI(false);
       }
     },
-    [selectedSafe, onContractABIChange],
+    [selectedVerifiedSafe, onContractABIChange],
   );
 
   const usefulMethods: AbiItemExtended[] = useMemo(
     () => getContractUsefulMethods(transactionValues.abi),
     [transactionValues.abi],
   );
 
+  const isSpecificError = (error: Message, comparison: MessageDescriptor) => {
+    return (
+      isMessageDescriptor(error) &&
+      error.defaultMessage === comparison.defaultMessage
+    );
+  };
+
   useEffect(() => {
     if (
       transactionValues.contract &&
-      selectedSafe &&
-      // only run effect if safe chain changes
-      prevSafeChainId !== selectedSafe.chainId
+      safe &&
+      // only run effect if safe chain changes or there was previously an error
+      (prevSafeChainId !== selectedVerifiedSafe?.chainId || ABIError)
```

Comment:
This fixes the bug where it wouldn't refetch if you had an invalid safe error and then updated the safe input